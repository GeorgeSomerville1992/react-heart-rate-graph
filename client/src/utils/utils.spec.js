import helper from './helpers';

const fixture = [
  {
		"minimum": "71",
		"maximum": "74",
		"meanAverage": "72",
		"dateTime": "2018-03-28T13:46:00"
	},
	{
		"minimum": "57",
		"maximum": "87",
		"meanAverage": "71",
		"dateTime": "2018-03-28T18:00:01"
	},
	{
		"minimum": "54",
		"maximum": "61",
		"meanAverage": "57",
		"dateTime": "2018-03-29T18:04:00"
	},
	{
		"minimum": "52",
		"maximum": "52",
		"meanAverage": "52",
		"dateTime": "2018-03-30T03:49:00"
	},
	{
		"minimum": "56",
		"maximum": "61",
		"meanAverage": "58",
		"dateTime": "2018-03-31T05:43:00"
	},
	{
		"minimum": "64",
		"maximum": "68",
		"meanAverage": "66",
		"dateTime": "2018-04-01T21:34:00"
  },
  {
		"minimum": "57",
		"maximum": "58",
		"meanAverage": "57",
		"dateTime": "2018-04-02T19:48:00"
	},
	{
		"minimum": "57",
		"maximum": "59",
		"meanAverage": "58",
		"dateTime": "2018-04-03T08:01:00"
	},
	{
		"minimum": "71",
		"maximum": "74",
		"meanAverage": "72",
		"dateTime": "2018-04-04T09:54:00"
	},
]

describe('Basic Utils tests', () => {
  it('convertValues: correctly converts values into the right format', () => {
    expect(helper.convertValues(fixture)).toMatchSnapshot()
  })

  it('getDateRange: returns a correct range', () => {
    const convertedFixture = helper.convertValues(fixture)

    expect(helper.getDateRange(convertedFixture)).toMatchSnapshot()
  })

  it('filterCurrentSelectedMonth: returns values based on a selcted month', () => {
    const convertedFixture = helper.convertValues(fixture)

    expect(helper.filterCurrentSelectedMonth(new Date("2018-04-01T21:34:00"),  convertedFixture)).toMatchSnapshot()
  })

  it('filterCurrentSelectedMonth: returns values based on a selcted date', () => {
    const convertedFixture = helper.convertValues(fixture)

    expect(helper.filterCurrentSelectedDate(new Date("2018-04-01T21:34:00"),  convertedFixture)).toMatchSnapshot()
  })
})