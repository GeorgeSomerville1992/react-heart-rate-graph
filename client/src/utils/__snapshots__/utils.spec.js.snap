// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Utils tests convertValues: correctly converts values into the right format 1`] = `
Array [
  Object {
    "dateTime": "2018-03-28 13:00",
    "maximum": 74,
    "meanAverage": 72,
    "minimum": 71,
  },
  Object {
    "dateTime": "2018-03-28 18:01",
    "maximum": 87,
    "meanAverage": 71,
    "minimum": 57,
  },
  Object {
    "dateTime": "2018-03-29 18:00",
    "maximum": 61,
    "meanAverage": 57,
    "minimum": 54,
  },
  Object {
    "dateTime": "2018-03-30 03:00",
    "maximum": 52,
    "meanAverage": 52,
    "minimum": 52,
  },
  Object {
    "dateTime": "2018-03-31 05:00",
    "maximum": 61,
    "meanAverage": 58,
    "minimum": 56,
  },
  Object {
    "dateTime": "2018-04-01 21:00",
    "maximum": 68,
    "meanAverage": 66,
    "minimum": 64,
  },
  Object {
    "dateTime": "2018-04-02 19:00",
    "maximum": 58,
    "meanAverage": 57,
    "minimum": 57,
  },
  Object {
    "dateTime": "2018-04-03 08:00",
    "maximum": 59,
    "meanAverage": 58,
    "minimum": 57,
  },
  Object {
    "dateTime": "2018-04-04 09:00",
    "maximum": 74,
    "meanAverage": 72,
    "minimum": 71,
  },
]
`;

exports[`Basic Utils tests filterCurrentSelectedMonth: returns values based on a selcted date 1`] = `
Array [
  Object {
    "dateTime": "2018-04-01 21:00",
    "maximum": 68,
    "meanAverage": 66,
    "minimum": 64,
  },
]
`;

exports[`Basic Utils tests filterCurrentSelectedMonth: returns values based on a selcted month 1`] = `
Array [
  Object {
    "dateTime": "2018-04-01 21:00",
    "maximum": 68,
    "meanAverage": 66,
    "minimum": 64,
  },
  Object {
    "dateTime": "2018-04-02 19:00",
    "maximum": 58,
    "meanAverage": 57,
    "minimum": 57,
  },
  Object {
    "dateTime": "2018-04-03 08:00",
    "maximum": 59,
    "meanAverage": 58,
    "minimum": 57,
  },
  Object {
    "dateTime": "2018-04-04 09:00",
    "maximum": 74,
    "meanAverage": 72,
    "minimum": 71,
  },
]
`;

exports[`Basic Utils tests getDateRange: returns a correct range 1`] = `
Array [
  2018-03-28T12:00:00.000Z,
  2018-04-04T08:00:00.000Z,
]
`;
